linode/manager
# refactor: [M3-4876] - LandingHeader & EntityHeader Refactor

## Description 📝
The `LandingHeader` has gone through a few iterations resulting in unused props and an excessively intricate composition with `EntityHeader`. Additionally, multiple header implementations on landing pages led to code duplication.

## Approach
To avoid one component handling too many tasks, we have simplified the components by decoupling `LandingHeader` and `EntityHeader`. Now, each page should have one `LandingHeader` along with several `EntityHeader`'s. Presently, `EntityHeader` is solely utilized for the Linode summary view, but I envision incorporating it for all entity headers in a separate PR. This was mainly to unify the headers across the app. We could revisit these components with UX to see how we can streamline things in the future.

## Major Changes
### Files for close review:
- `EntityHeader.tsx`
- `LandingHeader.tsx`
- `LinodeEntityDetail.tsx`
- `LinodeNetworking.tsx` - I removed the use of `EntityHeader` here because it was a one-off for this use-case. As I said, there are plans to reincorporate `EntityHeader` in a separate PR.
- `LinodeDetailHeader.tsx` - Copied over small parts from `LinodeDetailsBreadcrumb.tsx` before it was deleted.

### Added: 
- LandingHeader.stories.tsx
- EntityHeader.stories.tsx
    
### Removed
- `EntityHeader/HeaderBreadCrumb.tsx`: Variation of `Breadcrumb` that could be removed now.
- `LinodeDetailHeader/LinodeDetailsBreadcrumb.tsx`: Variation of `Breadcrumb` that could be removed now.
- `LinodeControls.tsx`: No longer used
- `MigrateLanding.tsx`: No longer used

### Fixes
- `IconTextLink.tsx`: This was never working, and essentially the math comes out to just calling `theme.spacing(1)`

## Preview 📷
![Screen Shot 2023-03-07 at 1 12 41 PM](https://user-images.githubusercontent.com/125309814/223911699-1fa8b8e6-384b-4ad2-886f-80d692071e45.png)
![Screen Shot 2023-03-07 at 1 13 06 PM](https://user-images.githubusercontent.com/125309814/223911702-ca51f76d-937f-46a3-a1b7-b9ef0f57963b.png)
![Screen Shot 2023-03-07 at 1 13 38 PM](https://user-images.githubusercontent.com/125309814/223911706-77bb5cec-1299-4e10-9c7c-007195152e96.png)
![Screen Shot 2023-03-07 at 1 14 02 PM](https://user-images.githubusercontent.com/125309814/223911708-2eda092b-08b5-40f7-bb28-b8437f72d15c.png)

## How to test 🧪
- Root app: `yarn dev`
- Root app: `yarn cy:run` for E2E tests
- App should run @ `http://localhost:3000/`
- Turn `Mock Service Workers` on using our DevTools (bottom left corner of window)

![Screen Shot 2023-03-08 at 10 54 14 PM](https://user-images.githubusercontent.com/125309814/223912710-540a3fd2-3224-4ade-9c36-267cd64e4bb8.png)
